src/main/java/com/shipmentEvents/handlers/EventHandler.java

```diff
@@ -1,195 +1,195 @@
-                lastKnownStatus = new MutablePair<Long, String>(timeStamp, status);
-                latestStatusForTrackingNumber.put(trackingNumber, lastKnownStatus);
-            }
+                String status = line1.split(":")[1];
+                Long timeStamp = Long.parseLong(line2.split(":")[1]);
+
+
+                if (null == lastKnownStatus || lastKnownStatus.getLeft() < timeStamp) {
+                    lastKnownStatus = new MutablePair<Long, String>(timeStamp, status);
+                    latestStatusForTrackingNumber.put(trackingNumber, lastKnownStatus);
+                }
 
-            //Add to list of processed files
-            filesProcessed.add(new KeyVersion(summary.getKey()));
-            logger.log("logging Contents of the file" + fileContents);
+                //Add to list of processed files
+                filesProcessed.add(new KeyVersion(summary.getKey()));
+                logger.log("logging Contents of the file" + fileContents);
+            }
+            return filesProcessed;
         }
-        return filesProcessed;
-    }
-    
-
-    private void deleteProcessedFiles(Map<String, List<KeyVersion>> filesToDelete) {
-      final AmazonS3 s3Client = EventHandler.getS3Client();
-      for (Entry<String, List<KeyVersion>> entry : filesToDelete.entrySet()) {
-          final DeleteObjectsRequest deleteRequest = new DeleteObjectsRequest(entry.getKey()).withKeys(entry.getValue()).withQuiet(false);
-          s3Client.deleteObjects(deleteRequest);
-      }
-    }
-    
-    private boolean isValidFile(String fileContents) {
-        if (!fileContents.contains("\n")) {
-            return false;
+
+
+        private void deleteProcessedFiles(Map<String, List<KeyVersion>> filesToDelete) {
+          final AmazonS3 s3Client = EventHandler.getS3Client();
+          for (Entry<String, List<KeyVersion>> entry : filesToDelete.entrySet()) {
+              final DeleteObjectsRequest deleteRequest = new DeleteObjectsRequest(entry.getKey()).withKeys(entry.getValue()).withQuiet(false);
+              s3Client.deleteObjects(deleteRequest);
+          }
         }
-        String[] lines = fileContents.split("\n");
-        for (String l: lines) {
-            if (!l.contains(":")) {
+
+        private boolean isValidFile(String fileContents) {
+            if (!fileContents.contains("\n")) {
                 return false;
             }
+            String[] lines = fileContents.split("\n");
+            for (String l: lines) {
+                if (!l.contains(":")) {
+                    return false;
+                }
+            }
+            return true;
         }
-        return true;
-    }
-    
-    public static AmazonS3 getS3Client() {
-        return AmazonS3ClientBuilder.standard().withRegion(Regions.DEFAULT_REGION).build();
+
+        public static AmazonS3 getS3Client() {
+            return AmazonS3ClientBuilder.standard().withRegion(Regions.DEFAULT_REGION).build();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code is written so that the client cannot be reused across invocations of the Lambda function.
To improve the performance of the Lambda function, consider using static initialization/constructor, global/static variables and singletons. It allows to keep alive and reuse HTTP connections that were established during a previous invocation.
Learn more about [best practices for working with AWS Lambda functions](https://docs.aws.amazon.com/lambda/latest/dg/best-practices.html).